release/release-debian

```diff
@@ -0,0 +1,145 @@
+#!/bin/bash -eu
+#
+# release-debian
+#
+# A release script that publishes packages to a debian repository.
+#
+# Arguments are described here. Most arguments have an equivalent envvar.
+#
+# -d src     RELEASE_DEBIAN_SOURCE=src  Debian source repository
+# -r repo    RELEASE_DEBIAN_REPO=repo   Generated Debian repository
+#
+# -q         RELEASE_QUIET=1            Make output more quiet
+# -x         RELEASE_TRANSACTION=1      SIGSTOP before pushing the dist-git commit
+# -v         RELEASE_VERBOSE=1          Make output more verbose
+# -z         RELEASE_CHECK=1            Check credentials only
+#
+
+set -eu
+
+TRANSACTION=${RELEASE_TRANSACTION:-0}
+CHECK=${RELEASE_CHECK:-0}
+QUIET=${RELEASE_QUIET:-0}
+VERBOSE=${RELEASE_VERBOSE:-0}
+
+TAG=${RELEASE_TAG:-}
+DEBIAN_SOURCE=${RELEASE_DEBIAN_SOURCE:-}
+DEBIAN_REPO=${RELEASE_DEBIAN_REPO:-debian-apt}
+
+usage()
+{
+    echo "usage: release-debian [-qvxz] [-t tag] [-d source] [-r repository]" >&2
+    exit ${1-2}
+}
+
+trace()
+{
+    if [ $QUIET -eq 0 ]; then
+        echo "> $@" >&2
+    fi
+}
+
+message()
+{
+    echo "release-debian: $@" >&2
+}
+
+check()
+{
+    true
+}
+
+prepare()
+{
+    trace "Creating Debian repository"
+
+    rm -rf debian
+    cp -r tools/debian .
+
+    printf "USENETWORK=yes\n" > pbuilderrc
+
+    sudo pbuilder clean
+    sudo pbuilder create --distribution unstable
```

Comment:
Because `docker build` doesn't allow privileged operations. If it did, we could move the "create" step into the Dockerfile. We'd still need to call `pbuilder update` here to make sure we're testing on latest unstable. I'm not sure if that would gain anything (depends on how much churn there is in the base packages).

Pbuilder always downloads and installs build dependencies when building a package.

We could mount /var/cache/pbuilder/aptcache from somewhere, so that we don't need to download the packages again after exiting the container.